rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java

```diff
@@ -851,6 +874,21 @@ public synchronized void registerLanguage(LanguageParameter lang) {
         if (client != null) {
             fact.setClient(client);
         }
+
+        // If we opened any files with this extension before, now associate them with contributions
+        var extensions = Arrays.asList(lang.getExtensions());
+        for (var e : files.entrySet()) {
+            var f = e.getKey();
+            var state = e.getValue();
+            if (extensions.contains(extension(f))) {
+                logger.trace("Open file of language {} - updating state: {}", lang.getName(), f);
+                var c = state.getCurrentContent();
+                state = files.replace(f, new TextDocumentState(contributions(f)::parsing, f, c.version(), state.getCurrentContent().get(), state.getLastModified()));
```

Comment:
I think we should have a function on `TextDocumentState`  called `changeParser` or something, such that it can be responsible for passing on the right state to the new instance?